src/vs/platform/dialogs/electron-main/dialogMainService.ts

```diff
@@ -48,14 +50,13 @@ export class DialogMainService implements IDialogMainService {
 
 	private static readonly workingDirPickerStorageKey = 'pickerWorkingDir';
 
-	private readonly mapWindowToDialogQueue: Map<number, Queue<void>>;
-	private readonly noWindowDialogQueue: Queue<void>;
+	private readonly windowDialogLocks = new Map<number, Set<number>>();
+	private readonly windowDialogQueues = new Map<number, Queue<any>>();
+	private readonly noWindowDialogueQueue = new Queue<any>();
```

Comment:
I know we had `any` before already but I would really like us to remove `any` and rather use the or type `MessageBoxReturnValue | SaveDialogReturnValue | OpenDialogReturnValue` here because that is what the queue is used for. This needs a few tweaks in `getWindowDialogQueue`, i.e. change that method to:

```ts
private getWindowDialogQueue<T extends MessageBoxReturnValue | SaveDialogReturnValue | OpenDialogReturnValue>(window?: BrowserWindow): Queue<T> {
```

but I think that makes it clear.

Reply:
Changed to what you've suggested in ae5d9e4